mmr/types.go

```diff
@@ -0,0 +1,17 @@
+package mmr
+
+type leaf struct {
+	pos  uint64
+	hash interface{}
+}
+
+type leafWithHash struct {
+	pos    uint64
+	hash   interface{}
+	height uint32
+}
+
+type peak struct {
+	height uint32
+	pos    uint64
+}
```

Comment:
Camelcase names?

Reply:
yes, camelcase names is the general practise in go